### PR TITLE
docs: build and push the API docs to github pages

### PR DESCRIPTION
**Pull request by elopio**
Requires #1125

### CODE REVIEW

File: docs/api/Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
```
Comment (sergiusens) [inline docs/api/Makefile]: why isn't this in `setup.py` instead?

Comment (elopio) [inline docs/api/Makefile]: This is autogenerated by sphinx. I'd prefer not to write anything by myself, and just use the tool.

**Comment by kyrofa** [comment]
CLA check here.

**Comment by elopio** [comment]
@kyrofa I think I'll have to copy the changes into a new branch, because I don't know where it took my personal email and I don't know how to force it to take my work email. I deleted the branch, rebased to make a new single commit, and made the pr again, and it still fails.

**Comment by kyrofa** [comment]
@elopio you probably made a `git config --global` with your personal details. Use `git config` within the snapcraft repo with your work details, then `git commit --amend --reset-author`.

**Comment by elopio** [comment]
Thanks @kyrofa, --reset-author was the magical step.

File: tools/push_api_gh_pages.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -ev
+
+GH_REPO="@github.com/elopio/snapcraft.git"
```
Comment (sergiusens) [inline tools/push_api_gh_pages.sh]: Can you push to a `gh-pages` branch on github.com:snapcore/snapcraft.git instead?

File: tools/push_api_gh_pages.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+FULL_REPO="https://$GH_TOKEN$GH_REPO"
+
+cd docs/api/_build/html
```
Comment (sergiusens) [inline tools/push_api_gh_pages.sh]: the plan was to delete `docs` and add `snappy-docs` as a submodule, can this be somewhere else?

**Comment by sergiusens** [comment]
oh, and there are conflicts :-)

File: tools/push_api_gh_pages.sh (hunk 2 above)
Comment (elopio) [inline tools/push_api_gh_pages.sh]: @sergiusens done. But then you need to update the token in the travis settings, with permissions to repo and user.

File: tools/push_api_gh_pages.sh (hunk 3 above)
Comment (elopio) [inline tools/push_api_gh_pages.sh]: I moved it to tools.

**Pull request merged**